### PR TITLE
adds a new magazine type exclusive to vector (take 2)

### PR DESCRIPTION
**Pull request by 9percents**
take 2, it works properly now
ammo380-E
:cl:
 * rscadd: Adds a new .380 magazine, holds 20 bullets, exclusive for vector. Can be made with ammolathe.

### CODE REVIEW

File: code/modules/projectiles/guns/projectile.dm [hunk 1]
```diff
@@ -19,6 +19,7 @@
 	var/obj/item/ammo_storage/magazine/stored_magazine = null
 	var/obj/item/ammo_casing/chambered = null
 	var/mag_type = ""
+	var/obj/item/ammo_storage/magazine/mag_type_restricted = null //better magazine manipulation
```
Comment (PaulRitter) [inline code/modules/projectiles/guns/projectile.dm]: any reason you would not make this a list of types?

Comment (PaulRitter) [reply code/modules/projectiles/guns/projectile.dm]: also, this should be `var/mag_type_restricted`

**Review by PaulRitter — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/projectile.dm (hunk 1 above)
Comment (9percents) [reply code/modules/projectiles/guns/projectile.dm]: it's a pretty niche variable overall, I don't think it'll see a lot of usage

File: code/modules/projectiles/guns/projectile.dm [hunk 2]
```diff
@@ -19,6 +19,7 @@
 	var/obj/item/ammo_storage/magazine/stored_magazine = null
 	var/obj/item/ammo_casing/chambered = null
 	var/mag_type = ""
+	var/list/mag_type_restricted() = list() //better magazine manipulation
```
Comment (PaulRitter) [inline code/modules/projectiles/guns/projectile.dm]: no `()`

File: code/modules/projectiles/guns/projectile.dm [hunk 3]
```diff
@@ -43,7 +44,7 @@
 
 //loads the argument magazine into the gun
 /obj/item/weapon/gun/projectile/proc/LoadMag(var/obj/item/ammo_storage/magazine/AM, var/mob/user)
-	if(istype(AM, text2path(mag_type)) && !stored_magazine)
+	if(istype(AM, text2path(mag_type)) && !stored_magazine && !istype(AM, mag_type_restricted))
```
Comment (PaulRitter) [inline code/modules/projectiles/guns/projectile.dm]: change this to work with a list, eg:
loop over contents and try an istype for each one

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 4]
```diff
@@ -146,6 +146,7 @@
 	icon_state = "secglockfancy"
 	ammo_type = "/obj/item/ammo_casing/c380auto"
 	mag_type = "/obj/item/ammo_storage/magazine/m380auto"
+	mag_type_restricted = /obj/item/ammo_storage/magazine/m380auto/extended
```
Comment (PaulRitter) [inline code/modules/projectiles/guns/projectile/pistol.dm]: `list(/obj/item/ammo_storage/magazine/m380auto/extended)`

**Review by PaulRitter — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
i know that feel 🔫

**Comment by 9percents** [comment]
done, I rebased so it doesn't have any conflicts with base branch
btw, 100% tested and ready to be pulled right now :^)

**Pull request merged**